### PR TITLE
Annotator module ts migration

### PR DESCRIPTION
**Pull request by acelaya**
This PR migrates a bunch of files from the annotator module to TypeScript.

A few were left out to make reviews easier. Continues on https://github.com/hypothesis/client/pull/5357

### CODE REVIEW

File: src/annotator/bucket-bar-client.ts [hunk 1]
```diff
@@ -1,35 +1,34 @@
...
  * 2. The position of anchors relative to the viewport of the guest has changed
- *
- * @implements {Destroyable}
  */
-export class BucketBarClient {
-  /**
-   * @param {object} options
-   *   @param {Element} options.contentContainer - The scrollable container element for the
-   *     document content. All of the highlights that the bucket bar's buckets point
-   *     at should be contained within this element.
```
Comment (lyzadanger) [inline src/annotator/bucket-bar-client.ts:26]: Looks like we lost this comment...

File: src/annotator/features.ts [hunk 2]
```diff
@@ -1,48 +1,38 @@
...
+  constructor(knownFlags: string[] = annotatorFlags) {
     super();
 
-    /**
-     * Map of flag name to enabled state.
-     *
-     * @type {Map<string, boolean>}
-     */
-    this._flags = new Map();
+    this._flags = new Map<string, boolean>();
```
Comment (lyzadanger) [inline src/annotator/features.ts:26]: This reminds me to ask! Where are we on making a decision about moving away from `_`-prefixed private attributes?

File: src/annotator/selection-observer.ts [hunk 3]
```diff
@@ -22,15 +19,21 @@ export function selectedRange(document) {
  * An observer that watches for and buffers changes to the document's current selection.
  */
 export class SelectionObserver {
+  private _pendingCallback: number | null;
```
Comment (lyzadanger) [inline src/annotator/selection-observer.ts:22]: Perhaps add a comment on this field: It's not obvious without reading code that it is used to hold a timeout ID. (I know there wasn't a comment before, either).

**Review by lyzadanger — APPROVED** [review]
Just a few comments!

File: src/annotator/features.ts (hunk 2 above)
Comment (acelaya) [reply src/annotator/features.ts:26]: There has been no decision as far as I know, so I have been keeping the same syntax.

**Pull request merged**